src/Helpers.php

```
@@ -156,4 +164,125 @@ public static function cleanSchemaName(string $name): string {
+	}
+
+	protected static function getScopeNameFromConst(ClassConstFetch $scope): string {
+		return match ($scope->name->name) {
+			'SCOPE_DEFAULT' => 'default',
+			'SCOPE_ADMINISTRATION' => 'administration',
+			'SCOPE_FEDERATION' => 'federation',
+			'SCOPE_IGNORE' => 'ignore',
+			// Fall back for future scopes assuming we follow the pattern (cut of 'SCOPE_' and lower case)
+			default => strtolower(substr($scope->name->name, 6)),
+		};
+	}
+
+	public static function getOpenAPIAttributeScopes(ClassMethod|Class_|Node $node, string $routeName): array {
+		$scopes = [];
+
+		/** @var AttributeGroup $attrGroup */
+		foreach ($node->attrGroups as $attrGroup) {
+			foreach ($attrGroup->attrs as $attr) {
+				if ($attr->name->getLast() === self::OPENAPI_ATTRIBUTE_CLASSNAME) {
+					if (empty($attr->args)) {
+						$scopes[] = 'default';
+						continue;
+					}
+
+					foreach ($attr->args as $key => $arg) {
+						$scope = self::getScopeNameFromAttributeArgument($arg, (int) $key, $routeName);
+						if ($scope !== null) {
+							$scopes[] = $scope;
+						}
+					}
+				}
+			}
+		}
+
+		return $scopes;
+	}
+
+	public static function getOpenAPIAttributeTagsByScope(ClassMethod|Class_|Node $node, string $routeName, string $defaultTag, string $defaultScope): array {
+		$tags = [];
+
+		/** @var AttributeGroup $attrGroup */
+		foreach ($node->attrGroups as $attrGroup) {
+			foreach ($attrGroup->attrs as $attr) {
+				if ($attr->name->getLast() === self::OPENAPI_ATTRIBUTE_CLASSNAME) {
+					if (empty($attr->args)) {
+						$tags[$defaultScope] = [$defaultTag];
+						continue;
+					}
+
+					$foundTags = [];
+					$foundScopeName = null;
+					foreach ($attr->args as $key => $arg) {
+						$foundScopeName = self::getScopeNameFromAttributeArgument($arg, (int) $key, $routeName);
+
+						if ($arg->name?->name !== 'tags' && ($arg->name !== null || $key !== 1)) {
+							continue;
+						}
+
+						if (!$arg->value instanceof Array_) {
+
+							continue;
+						}
```

This should panic instead of ignoring wrong values